src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let switcher = true;
+
+    while (switcher) {
+      switcher = false;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const firstValue = this[i];
+        const secondValue = this[i + 1];
+        const compareString = (a, b) => (String(a) > String(b) ? 1 : -1);
```

Comment:
The `compareString` function is created inside the loop on every iteration. This is inefficient as the function can be created once and reused. Consider moving the `compareString` function outside of the loop to avoid unnecessary function creation on each iteration.